webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -975,6 +1092,42 @@ public static ByteArrayOutputStream requestsToZip(ServletContext context, String
 
         ZipOutputStream zipOut = new ZipOutputStream(baosZip);
         HashMap<String, Boolean> seenTitles = new HashMap<String, Boolean>();
+
+        for(ScrambleRequest scrambleRequest : scrambleRequests) {
+            if (scrambleRequest.fmc) {
```

Comment:
nit: I would probably create a boolean called `isFmcBeingHeld` and update it here and then immediately break.

Afterwards, I'd create an `if (isFmcBeingHeld)` block and put all the logic for creating the generic FMC solution sheet there. This has a benefit of decreasing indentation a bit, and making it very clear that we aren't going to create multiple copies of the same `pdf/3x3x3 Fewest Moves Solution Sheet.pdf` file. With the current code, I had to search a bit to find the `break;` at the end of this block.

Just a comment on how I would do it, feel free to ignore!